Shopify/shopify-cli
# Add support for script metaobject

### WHY are these changes introduced?

Fixes https://github.com/Shopify/script-service/issues/4815 



We want to replace the `script.config.yml` with some file containing a metaobject definition. For this PR it is called `function.metaobject.yml` but the file name and format are very much TBD (but cheap to change).

### WHAT is this pull request doing?



This PR now requires that a script contains a `function.metaobject.yml` file, and sends it to the server during push, where it is validated and the errors are displayed here in the CLI.

### How to test your changes?



Use my spin instance: workspace is `script-service-x3mn`.

Create a new script with the `--branch=ap.metaobject` flag to get the `function.metaobject.yml` file in your new script. Push the script and see that it succeeds. Make a change to the metaobject definition in `function.metaobject.yml` so that it fails validation, push and see an error. Remove the `function.metaobject`, push and see an error. Put invalid YAML in the file, push and see an error.

### Update checklist
